mbed-client-pal/Source/Port/Reference-Impl/OS_Specific/ZephyrOS/RTOS/pal_plat_rtos.c

```diff
@@ -181,27 +181,28 @@ palStatus_t pal_plat_osDelay(uint32_t milliseconds)
 }
 
 struct timer_data {
-    struct k_work_delayable work;
+    struct k_work_delayable dwork;
+    struct k_work_sync work_sync;
     palTimerFuncPtr callback;
     void *arg;
     struct k_mutex lock;
-    struct k_sem busy;
-    uint64_t timeout;
-    uint64_t period;
+    int64_t timeout;
+    int32_t period;
 };
 
 static void work_fn(struct k_work *work)
 {
     struct k_work_delayable *dwork = CONTAINER_OF(work, struct k_work_delayable, work);
-    struct timer_data *timer_data = CONTAINER_OF(dwork, struct timer_data, work);
+    struct timer_data *timer_data = CONTAINER_OF(dwork, struct timer_data, dwork);
 
-    uint64_t uptime = k_uptime_get();
+    int64_t uptime = k_uptime_get();
+    int ret = 0;
 
     k_mutex_lock(&timer_data->lock, K_FOREVER);
 
-    uint64_t timeout = timer_data->timeout;
+    int64_t timeout = timer_data->timeout;
```

Comment:
why do you use int64_t for timeout?

Reply:
I don't think this can be negative and there can be an overflow issue

Reply:
Nothing besides i wanted the type to match to what is returned by `k_uptime_get()`. I'll revert this change.